File: feature-favorites/src/main/java/com/andyanika/translator/feature/favorites/FavoritesViewHolder.java
package com.andyanika.translator.feature.favorites;

import androidx.recyclerview.widget.RecyclerView;
import android.view.View;
import android.widget.ImageButton;
import android.widget.TextView;

import com.andyanika.translator.common.models.FavoriteModel;
import com.andyanika.translator.common.models.TranslateResult;

import io.reactivex.rxjava3.core.Observer;

public class FavoritesViewHolder extends RecyclerView.ViewHolder {
    private TextView title;
    private TextView description;
    private TextView langSrc;
    private TextView langDst;

    FavoritesViewHolder(View itemView, final Observer<Integer> clickObserver) {
        super(itemView);
        title = itemView.findViewById(R.id.title);
        description = itemView.findViewById(R.id.description);
        langSrc = itemView.findViewById(R.id.txt_lang_src);
        langDst = itemView.findViewById(R.id.txt_lang_dst);
        ImageButton favoriteButton = itemView.findViewById(R.id.btn_favorite);
        favoriteButton.setImageResource(R.drawable.ic_close);
        favoriteButton.setOnClickListener(v -> clickObserver.onNext(getAdapterPosition()));
    }

    public void bind(FavoriteModel model) {
        TranslateResult result = model.translateResult;

        title.setText(result.textSrc);
        description.setText(result.textDst);
        langSrc.setText(result.direction.getSrc().toString());
        langDst.setText(result.direction.getDst().toString());
    }
}

File: feature-favorites/src/main/kotlin/com/andyanika/translator/feature/favorites/FavoritesViewHolder.kt
package com.andyanika.translator.feature.favorites

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.andyanika.translator.feature.favorites.databinding.ListviewRowBinding
import core.models.FavoriteModel

internal class FavoritesViewHolder(
    itemView: View,
    private val action: (FavoriteModel) -> Unit
) : RecyclerView.ViewHolder(itemView) {

    fun bind(model: FavoriteModel) = with(ListviewRowBinding.bind(itemView)) {
        val (textSrc, textDst, direction) = model.translateResult

        title.text = textSrc
        description.text = textDst
        txtLangSrc.text = direction.src.toString()
        txtLangDst.text = direction.dst.toString()
        btnFavorite.apply {
            setImageResource(R.drawable.ic_close)
            setOnClickListener { action.invoke(model) }
        }
    }
}
